src/components/common/DiscountCodes.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+    <div class="row">
+      <div class="col-sm-12">
+        <div id="appliedDiscounts"
+             class="text-right col-sm-12">
+          {{ $t('appliedDiscounts') }}:
+        </div>
+        <div v-for="discountInfo in cartLike.discountCodes"
+             :key='discountInfo.discountCode.id'
+             class="text-right col-sm-offset-6"
+             data-test="discount-code-list">
+          <div id="discountCodeName"
+               class="col-sm-8"
+               data-test="discount-code-name">
+            {{ discountInfo.discountCode.code }}
+          </div>
+          <div class="col-sm-3">{{ discountInfo.discountCode.name }}</div>
+          <RemoveDiscountCode :editable='true'
```

Comment:
RemoveDiscountCode is only going to be used when it is editable, so it should not receive an "editable" flag. It should be the parent component deciding whether to invoke the RemoveDiscountCode depending on whether is editable or not.

At the moment, you are calling this component inside myOrders, even though it shouldn't. And my guess is that it's even showing you the possibility of removing discount codes.

Reply:
so I guess the `DiscountCodes.vue` should receive 'editable' flag and basing on that show the remove component?

Reply:
Yep, exactly. It's that, or find out from within `DiscountCodes.vue` whether the `cartLike` is an `order` or a `cart`. The second one is probably cleaner.